src/js/index.js

```diff
@@ -22,26 +15,72 @@ const compare = (current, next) => {
 };
 const printAllTasks = () => {
   const taskList = document.getElementById('taskList');
-  const sorted = taskListArr.sort(compare);
+  taskList.innerHTML = '';
+  const sorted = taskmanager.taskList.sort(compare);
   sorted.forEach((task) => {
     const listViewItem = document.createElement('li');
     listViewItem.className = 'list-group-item d-flex justify-content-between align-items-center';
-
     const checkbox = document.createElement('input');
     checkbox.type = 'checkbox';
     checkbox.name = 'name';
 
     checkbox.className = 'form-check-input pull-left';
     checkbox.style.marginRight = '17px';
     checkbox.checked = task.completed;
+    checkbox.addEventListener('change', (event) => {
+      if (event.currentTarget.checked) {
+        task.updateStatus(true);
+      } else {
+        task.updateStatus(false);
+      }
+      taskmanager.editTask(task);
+    });
 
     const span = document.createElement('span');
     span.className = 'fas fa-ellipsis-v pull-right';
     listViewItem.appendChild(checkbox);
-    listViewItem.appendChild(document.createTextNode(task.description));
+    const ptag = document.createElement('span');
+    ptag.setAttribute('contenteditable', 'true');
+    ptag.setAttribute('class', 'single-line');
+    ptag.appendChild(document.createTextNode(task.description));
+    ptag.addEventListener('keypress', (event) => {
+      if (event.key === 'Enter') {
+        task.updateDescription(ptag.innerText);
+        taskmanager.editTask(task);
+        ptag.setAttribute('contenteditable', 'false');
+        ptag.setAttribute('contenteditable', 'true');
+      }
+    });
+    listViewItem.appendChild(ptag);
     listViewItem.appendChild(span);
     taskList.appendChild(listViewItem);
   });
+
+  // Add a new task
+  document.getElementById('newTask').addEventListener('keypress', (event) => {
+    if (event.key === 'Enter') {
+      const description = document.getElementById('newTask').value;
+      event.preventDefault();
+      if (description.length > 0) {
+        const newTodo = new Todo(description, false, 0);
```

Comment:
- [ ] When a task is added its index value should be set to the new length of the array. So if I add the first task its `index` should be 1, the second one should be 2,...the seventh one should be 7, and so on

Reply:
This handled in the taskManager